### PR TITLE
#3503 Add xslt.features support

### PR DESCRIPTION
**Pull request by DenisIstomin**
Related to #3503

### CODE REVIEW

File: integration-tests/xml/src/main/resources/application.properties [hunk 1]
```diff
@@ -24,3 +24,4 @@ quarkus.log.category."org.apache.camel.quarkus.core.deployment".level = INFO
 # Quarkus - Camel
 #
 quarkus.camel.xslt.sources = xslt/classpath-transform.xsl,xslt/html-transform.xsl,xslt/html-to-text.xsl
+quarkus.camel.xslt.features."http\://javax.xml.XMLConstants/feature/secure-processing" = true
```
Comment (ppalaga) [inline integration-tests/xml/src/main/resources/application.properties:27]: This change is not necessary because `true` is the default, no?

**Review by ppalaga — CHANGES_REQUESTED** [review]
Thanks, @DenisIstomin this looks quite good! 

Except for the inline suggestions, could you please add some test coverage to the test module under `integration-tests/xml`? Ideally a new XSLT file. If the functions implementations really have to be in a separate jar file, the you can create a new module under `/integration-tests-support`.

File: extensions/xslt/runtime/src/main/java/org/apache/camel/quarkus/component/xslt/CamelXsltRecorder.java [hunk 2]
```diff
@@ -52,16 +59,27 @@ public void addRuntimeUriResolverEntry(RuntimeValue<RuntimeUriResolver.Builder>
...
 
         @Override
         public void configure(TransformerFactory tf, XsltEndpoint endpoint) {
             final String className = uriResolver.getTransletClassName(endpoint.getResourceUri());
 
+            for (Map.Entry<String, Boolean> entry : features.entrySet()) {
+                try {
+                    tf.setFeature(entry.getKey(), entry.getValue());
+                } catch (TransformerException e) {
+                    LOGGER.warn("Unsupported TransformerFactory feature {}", entry.getKey());
```
[the remark this answers is not in the document]
Comment (DenisIstomin) [reply extensions/xslt/runtime/src/main/java/org/apache/camel/quarkus/component/xslt/CamelXsltRecorder.java]: Thanks, fixed

**Comment by DenisIstomin** [comment]
Thanks @ppalaga, added an integration test for the extension functions.

**Comment by ppalaga** [comment]
Great work, thanks @DenisIstomin !

I am checking locally, whether I can make loading the functions from the current project work. 
FWIW, it works in native mode because the classloader there is flat.

I have sent https://github.com/apache/camel-quarkus/pull/3533 to unblock building this and other PRs

**Comment by ppalaga** [comment]
> I am checking locally, whether I can make loading the functions from the current project work.
> FWIW, it works in native mode because the classloader there is flat.

I think this is caused by the fact that TransformerFactory class is resolved using a wrong class loader around here in Camel https://github.com/apache/camel/blob/camel-3.15.0/components/camel-xslt/src/main/java/org/apache/camel/component/xslt/XsltEndpoint.java#L372

I'd say this PR is good to go like this with the known limitation and I'll file a followup issue for the class loading problem.

**Comment by ppalaga** [comment]
The CI build should work now. Could you please squash your two commits, rebase on top of the recent master and force-push? - That should re-trigger the CI for this PR.

**Review by ppalaga — APPROVED** [review]
Thanks again, @DenisIstomin !

**Pull request merged**